kotlin/src/main/kotlin/com/linecorp/armeria/internal/common/kotlin/CoroutineUtil.kt

```diff
@@ -36,11 +37,12 @@ fun callKotlinSuspendingMethod(
     method: Method,
     obj: Any,
     args: Array<Any>,
+    executorService: ExecutorService,
     ctx: RequestContext
 ): CompletableFuture<Any?> {
     val kFunction = checkNotNull(method.kotlinFunction) { "method is not a suspending function" }
-    val coroutineContext = CoroutineContexts.getCoroutineContext(ctx)
-    val newContext = Dispatchers.Unconfined + (coroutineContext ?: EmptyCoroutineContext)
+    val coroutineContext = CoroutineContexts.getCoroutineContext(ctx) ?: EmptyCoroutineContext
+    val newContext = executorService.asCoroutineDispatcher() + coroutineContext
```

Comment:
Question: Is it possible to cache `executorService.asCoroutineDispatcher()`?

Reply:
For example, we can
- make a constant for a coroutine dispatcher of armeria event loop (which is not context-aware)
- and `ctx.push` at the dispatched thread before executing controller methods

Reply:
Yes.

Reply:
For example...

```kotlin
import com.linecorp.armeria.common.CommonPools
import com.linecorp.armeria.common.RequestContext
import com.linecorp.armeria.common.kotlin.CoroutineContexts
import com.linecorp.armeria.common.util.SafeCloseable
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.ThreadContextElement
import kotlinx.coroutines.asCoroutineDispatcher
import kotlinx.coroutines.future.future
import java.lang.reflect.Method
import java.util.concurrent.CompletableFuture
import kotlin.coroutines.AbstractCoroutineContextElement
import kotlin.coroutines.CoroutineContext
import kotlin.coroutines.EmptyCoroutineContext
import kotlin.reflect.full.callSuspend
import kotlin.reflect.jvm.kotlinFunction

val armeriaDispatcher = CommonPools.workerGroup().asCoroutineDispatcher()

class ArmeriaRequestContext(
    private val requestContext: RequestContext
) : ThreadContextElement<SafeCloseable>, AbstractCoroutineContextElement(Key) {

    companion object Key : CoroutineContext.Key<ArmeriaRequestContext>

    override fun updateThreadContext(context: CoroutineContext): SafeCloseable {
        return requestContext.push()
    }

    override fun restoreThreadContext(context: CoroutineContext, oldState: SafeCloseable) {
        oldState.close()
    }
}

/**
 * Invokes a suspending function and returns [CompletableFuture].
 */
fun callKotlinSuspendingMethod(
    method: Method,
    obj: Any,
    args: Array<Any>,
    ctx: RequestContext
): CompletableFuture<Any?> {
    val kFunction = checkNotNull(method.kotlinFunction) { "method is not a suspending function" }
    val coroutineContext = CoroutineContexts.get(ctx) ?: EmptyCoroutineContext
    // if `coroutineContext` contains a coroutine dispatcher, executorService is not used.
    val newContext = armeriaDispatcher + ArmeriaRequestContext(ctx) + coroutineContext
    return GlobalScope.future(newContext) {
        kFunction
            .callSuspend(obj, *args)
            .let { if (it == Unit) null else it }
    }
}
```

Reply:
Is this addressed? What's the result? 🤔

Reply:
I don't add this change yet.

Reply:
Ah, I think we can make a separate PR for this if you want. 😄

Reply:
Thanks. I want to separate this change from this PR.